--- a/src/jwt.cc	2023-09-05 23:03:52.783122389 +1000
+++ b/src/jwt.cc	2023-09-08 22:23:04.842969505 +1000
@@ -48,7 +48,10 @@
 }
 
 Status Jwt::parseFromString(const std::string& jwt) {
-  // jwt must have exactly 2 dots with 3 sections.
+  // jwt must have exactly 2 dots
+  if (std::count(jwt.begin(), jwt.end(), '.') != 2) {
+    return Status::JwtBadFormat;
+  }
   jwt_ = jwt;
   std::vector<absl::string_view> jwt_split =
       absl::StrSplit(jwt, '.', absl::SkipEmpty());
