--- a/include/openssl/base64.h
+++ b/include/openssl/base64.h
@@ -54,14 +54,14 @@
  * copied and put under another distribution licence
  * [including the GNU Public Licence.] */
 
-// #ifndef OPENSSL_HEADER_BASE64_H
-// #define OPENSSL_HEADER_BASE64_H
+#ifndef OPENSSL_HEADER_BASE64_H
+#define OPENSSL_HEADER_BASE64_H
 
-// #include <openssl/base.h>
+#include <openssl/base.h>
 
-// #if defined(__cplusplus)
-// extern "C" {
-// #endif
+#if defined(__cplusplus)
+extern "C" {
+#endif
 
 
 // base64 functions.
@@ -94,15 +94,15 @@
 // be needed to call |EVP_DecodeBase64| on an input of length |len|. It returns
 // one on success or zero if |len| is not a valid length for a base64-encoded
 // string.
-// OPENSSL_EXPORT int EVP_DecodedLength(size_t *out_len, size_t len);
+OPENSSL_EXPORT int EVP_DecodedLength(size_t *out_len, size_t len);
 
 // EVP_DecodeBase64 decodes |in_len| bytes from base64 and writes
 // |*out_len| bytes to |out|. |max_out| is the size of the output
 // buffer. If it is not enough for the maximum output size, the
 // operation fails. It returns one on success or zero on error.
-// OPENSSL_EXPORT int EVP_DecodeBase64(uint8_t *out, size_t *out_len,
-//                                     size_t max_out, const uint8_t *in,
-//                                     size_t in_len);
+OPENSSL_EXPORT int EVP_DecodeBase64(uint8_t *out, size_t *out_len,
+                                    size_t max_out, const uint8_t *in,
+                                    size_t in_len);
 
 
 // Deprecated functions.
@@ -191,8 +191,8 @@
 // };
 
 
-// #if defined(__cplusplus)
-// }  // extern C
-// #endif
+#if defined(__cplusplus)
+}  // extern C
+#endif
 
-// #endif  // OPENSSL_HEADER_BASE64_H
+#endif  // OPENSSL_HEADER_BASE64_H
