--- a/include/openssl/pkcs8.h
+++ b/include/openssl/pkcs8.h
@@ -54,16 +54,16 @@
  * Hudson (tjh@cryptsoft.com). */
 
 
-// #ifndef OPENSSL_HEADER_PKCS8_H
-// #define OPENSSL_HEADER_PKCS8_H
+#ifndef OPENSSL_HEADER_PKCS8_H
+#define OPENSSL_HEADER_PKCS8_H
 
-// #include <openssl/base.h>
-// #include <openssl/x509.h>
+#include <openssl/base.h>
+#include <openssl/x509.h>
 
 
-// #if defined(__cplusplus)
-// extern "C" {
-// #endif
+#if defined(__cplusplus)
+extern "C" {
+#endif
 
 
 // PKCS8_encrypt serializes and encrypts a PKCS8_PRIV_KEY_INFO with PBES1 or
@@ -124,9 +124,9 @@
 // success and zero on error. The caller takes ownership of the outputs.
 // Any friendlyName attributes (RFC 2985) in the PKCS#12 structure will be
 // returned on the |X509| objects as aliases. See also |X509_alias_get0|.
-// OPENSSL_EXPORT int PKCS12_get_key_and_certs(EVP_PKEY **out_key,
-//                                             STACK_OF(X509) *out_certs,
-//                                             CBS *in, const char *password);
+OPENSSL_EXPORT int PKCS12_get_key_and_certs(EVP_PKEY **out_key,
+                                            STACK_OF(X509) *out_certs,
+                                            CBS *in, const char *password);
 
 
 // Deprecated functions.
@@ -149,7 +149,7 @@
 //                                   size_t ber_len);
 
 // d2i_PKCS12_bio acts like |d2i_PKCS12| but reads from a |BIO|.
-// OPENSSL_EXPORT PKCS12* d2i_PKCS12_bio(BIO *bio, PKCS12 **out_p12);
+OPENSSL_EXPORT PKCS12* d2i_PKCS12_bio(BIO *bio, PKCS12 **out_p12);
 
 // d2i_PKCS12_fp acts like |d2i_PKCS12| but reads from a |FILE|.
 // OPENSSL_EXPORT PKCS12* d2i_PKCS12_fp(FILE *fp, PKCS12 **out_p12);
@@ -184,9 +184,9 @@
 // It returns one on success and zero on error.
 //
 // Use |PKCS12_get_key_and_certs| instead.
-// OPENSSL_EXPORT int PKCS12_parse(const PKCS12 *p12, const char *password,
-//                                 EVP_PKEY **out_pkey, X509 **out_cert,
-//                                 STACK_OF(X509) **out_ca_certs);
+OPENSSL_EXPORT int PKCS12_parse(const PKCS12 *p12, const char *password,
+                                EVP_PKEY **out_pkey, X509 **out_cert,
+                                STACK_OF(X509) **out_ca_certs);
 
 // PKCS12_verify_mac returns one if |password| is a valid password for |p12|
 // and zero otherwise. Since |PKCS12_parse| doesn't take a length parameter,
@@ -196,8 +196,8 @@
 // |password[password_len]| must be zero and no other NUL bytes may appear in
 // |password|. If the |password_len| checks fail, zero is returned
 // immediately.
-// OPENSSL_EXPORT int PKCS12_verify_mac(const PKCS12 *p12, const char *password,
-//                                      int password_len);
+OPENSSL_EXPORT int PKCS12_verify_mac(const PKCS12 *p12, const char *password,
+                                     int password_len);
 
 // PKCS12_DEFAULT_ITER is the default number of KDF iterations used when
 // creating a |PKCS12| object.
@@ -233,24 +233,24 @@
 //                                      int mac_iterations, int key_type);
 
 // PKCS12_free frees |p12| and its contents.
-// OPENSSL_EXPORT void PKCS12_free(PKCS12 *p12);
+OPENSSL_EXPORT void PKCS12_free(PKCS12 *p12);
 
 
-// #if defined(__cplusplus)
-// }  // extern C
+#if defined(__cplusplus)
+}  // extern C
 
-// extern "C++" {
+extern "C++" {
 
-// BSSL_NAMESPACE_BEGIN
+BSSL_NAMESPACE_BEGIN
 
-// BORINGSSL_MAKE_DELETER(PKCS12, PKCS12_free)
+BORINGSSL_MAKE_DELETER(PKCS12, PKCS12_free)
 // BORINGSSL_MAKE_DELETER(PKCS8_PRIV_KEY_INFO, PKCS8_PRIV_KEY_INFO_free)
 
-// BSSL_NAMESPACE_END
+BSSL_NAMESPACE_END
 
-// }  // extern C++
+}  // extern C++
 
-// #endif
+#endif
 
 #ifdef ossl_PKCS8_R_BAD_PKCS12_DATA
 #define PKCS8_R_BAD_PKCS12_DATA ossl_PKCS8_R_BAD_PKCS12_DATA
@@ -355,4 +355,4 @@
 #define PKCS8_R_AMBIGUOUS_FRIENDLY_NAME ossl_PKCS8_R_AMBIGUOUS_FRIENDLY_NAME
 #endif
 
-// #endif  // OPENSSL_HEADER_PKCS8_H
+#endif  // OPENSSL_HEADER_PKCS8_H
