--- a/include/openssl/rand.h
+++ b/include/openssl/rand.h
@@ -12,21 +12,21 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE. */
 
-// #ifndef OPENSSL_HEADER_RAND_H
-// #define OPENSSL_HEADER_RAND_H
+#ifndef OPENSSL_HEADER_RAND_H
+#define OPENSSL_HEADER_RAND_H
 
-// #include <openssl/base.h>
+#include <openssl/base.h>
 
-// #if defined(__cplusplus)
-// extern "C" {
-// #endif
+#if defined(__cplusplus)
+extern "C" {
+#endif
 
 
 // Random number generation.
 
 
 // RAND_bytes writes |len| bytes of random data to |buf| and returns one.
-// OPENSSL_EXPORT int RAND_bytes(uint8_t *buf, size_t len);
+OPENSSL_EXPORT int RAND_bytes(uint8_t *buf, size_t len);
 
 // RAND_cleanup frees any resources used by the RNG. This is not safe if other
 // threads might still be calling |RAND_bytes|.
@@ -107,8 +107,8 @@
 // OPENSSL_EXPORT int RAND_set_rand_method(const RAND_METHOD *);
 
 
-// #if defined(__cplusplus)
-// }  // extern C
-// #endif
+#if defined(__cplusplus)
+}  // extern C
+#endif
 
-// #endif  // OPENSSL_HEADER_RAND_H
+#endif  // OPENSSL_HEADER_RAND_H
