--- a/include/openssl/x509v3.h
+++ b/include/openssl/x509v3.h
@@ -52,17 +52,19 @@
  * (eay@cryptsoft.com).  This product includes software written by Tim
  * Hudson (tjh@cryptsoft.com). */
 
-// #ifndef HEADER_X509V3_H
-// #define HEADER_X509V3_H
+#ifndef HEADER_X509V3_H
+#define HEADER_X509V3_H
 
-// #include <openssl/bio.h>
-// #include <openssl/conf.h>
-// #include <openssl/lhash.h>
-// #include <openssl/x509.h>
-
-// #if defined(__cplusplus)
-// extern "C" {
-// #endif
+#include <openssl/bio.h>
+#include <openssl/conf.h>
+#include <openssl/lhash.h>
+#include <openssl/x509.h>
+
+#include <ossl/openssl/x509v3.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
 
 
 // Legacy X.509 library.
@@ -219,6 +221,7 @@
 //     ASN1_TYPE *other;       // x400Address
 //   } d;
 // } GENERAL_NAME;
+typedef struct ossl_GENERAL_NAME_st GENERAL_NAME;
 
 // DEFINE_STACK_OF(GENERAL_NAME)
 
@@ -497,7 +500,7 @@
 
 // TODO(https://crbug.com/boringssl/407): This is not const because it contains
 // an |X509_NAME|.
-// DECLARE_ASN1_FUNCTIONS(GENERAL_NAME)
+DECLARE_ASN1_FUNCTIONS(GENERAL_NAME)
 // OPENSSL_EXPORT GENERAL_NAME *GENERAL_NAME_dup(GENERAL_NAME *a);
 
 // GENERAL_NAME_cmp returns zero if |a| and |b| are equal and a non-zero
@@ -857,7 +860,7 @@
 // OPENSSL_EXPORT int X509_check_issued(X509 *issuer, X509 *subject);
 // OPENSSL_EXPORT int X509_check_akid(X509 *issuer, AUTHORITY_KEYID *akid);
 
-// OPENSSL_EXPORT uint32_t X509_get_extension_flags(X509 *x);
+OPENSSL_EXPORT uint32_t X509_get_extension_flags(X509 *x);
 // OPENSSL_EXPORT uint32_t X509_get_key_usage(X509 *x);
 // OPENSSL_EXPORT uint32_t X509_get_extended_key_usage(X509 *x);
 
@@ -958,12 +961,12 @@
 // made after this point may be overwritten when the script is next run.
 
 
-// #if defined(__cplusplus)
-// }  // extern C
+#if defined(__cplusplus)
+}  // extern C
 
-// extern "C++" {
+extern "C++" {
 
-// BSSL_NAMESPACE_BEGIN
+BSSL_NAMESPACE_BEGIN
 
 // BORINGSSL_MAKE_DELETER(ACCESS_DESCRIPTION, ACCESS_DESCRIPTION_free)
 // BORINGSSL_MAKE_DELETER(AUTHORITY_KEYID, AUTHORITY_KEYID_free)
@@ -971,16 +974,16 @@
 // TODO(davidben): Move this to conf.h and rename to CONF_VALUE_free.
 // BORINGSSL_MAKE_DELETER(CONF_VALUE, X509V3_conf_free)
 // BORINGSSL_MAKE_DELETER(DIST_POINT, DIST_POINT_free)
-// BORINGSSL_MAKE_DELETER(GENERAL_NAME, GENERAL_NAME_free)
+BORINGSSL_MAKE_DELETER(GENERAL_NAME, GENERAL_NAME_free)
 // BORINGSSL_MAKE_DELETER(GENERAL_SUBTREE, GENERAL_SUBTREE_free)
 // BORINGSSL_MAKE_DELETER(NAME_CONSTRAINTS, NAME_CONSTRAINTS_free)
 // BORINGSSL_MAKE_DELETER(POLICY_MAPPING, POLICY_MAPPING_free)
 // BORINGSSL_MAKE_DELETER(POLICYINFO, POLICYINFO_free)
 
-// BSSL_NAMESPACE_END
+BSSL_NAMESPACE_END
 
-// }  // extern C++
-// #endif
+}  // extern C++
+#endif
 
 #ifdef ossl_X509V3_R_BAD_IP_ADDRESS
 #define X509V3_R_BAD_IP_ADDRESS ossl_X509V3_R_BAD_IP_ADDRESS
@@ -1178,4 +1181,4 @@
 #define X509V3_R_TRAILING_DATA_IN_EXTENSION ossl_X509V3_R_TRAILING_DATA_IN_EXTENSION
 #endif
 
-// #endif
+#endif
