--- a/bazel/repositories.bzl
+++ b/bazel/repositories.bzl
@@ -233,18 +233,13 @@ def envoy_dependencies(skip_targets = []):
     # Setup external Bazel rules
     _foreign_cc_dependencies()
 
-    # Binding to an alias pointing to the selected version of BoringSSL:
-    # - BoringSSL FIPS from @boringssl_fips//:ssl,
-    # - non-FIPS BoringSSL from @boringssl//:ssl.
-    _boringssl()
-    _boringssl_fips()
     native.bind(
         name = "ssl",
-        actual = "@envoy//bazel:boringssl",
+        actual = "@bssl-compat//:bssl-compat",
     )
     native.bind(
         name = "crypto",
-        actual = "@envoy//bazel:boringcrypto",
+        actual = "@bssl-compat//:bssl-compat",
     )
 
     # The long repo names (`com_github_fmtlib_fmt` instead of `fmtlib`) are
@@ -1162,7 +1157,16 @@ def _emsdk():
     )
 
 def _com_github_google_jwt_verify():
-    external_http_archive("com_github_google_jwt_verify")
+    external_http_archive(
+        name = "com_github_google_jwt_verify",
+        patch_args = ["-p1"],
+        patches = [
+            "@envoy_openssl//:patch/jwt_verify_lib/src/jwks.cc.patch",
+            "@envoy_openssl//:patch/jwt_verify_lib/src/jwt.cc.patch",
+            "@envoy_openssl//:patch/jwt_verify_lib/src/status.cc.patch",
+            "@envoy_openssl//:patch/jwt_verify_lib/src/verify.cc.patch",
+        ]
+    )
 
     native.bind(
         name = "jwt_verify_lib",

